superset-frontend/plugins/plugin-chart-echarts/src/Timeseries/transformers.ts

```diff
@@ -559,6 +560,24 @@ export function getPadding(
   const yAxisOffset = addYAxisTitleOffset
     ? TIMESERIES_CONSTANTS.yAxisLabelTopOffset
     : 0;
+  const yAxisLeftOffset = showAxis
+    ? 0
+    : TIMESERIES_CONSTANTS.yAxisLabelLeftOffset;
+  let leftOffset: number;
+  if (yAxisTitlePosition === 'Left' && showAxis) {
+    leftOffset =
+      TIMESERIES_CONSTANTS.gridOffsetLeft + (Number(yAxisTitleMargin) || 0);
+  } else if (yAxisTitlePosition === 'Left' && !showAxis) {
+    leftOffset =
+      TIMESERIES_CONSTANTS.gridOffsetLeft +
+      (Number(yAxisTitleMargin) || 0) -
+      (Number(yAxisLeftOffset) || 0);
+  } else if (yAxisTitlePosition !== 'Left' && !showAxis) {
+    leftOffset =
+      TIMESERIES_CONSTANTS.gridOffsetLeft - (Number(yAxisLeftOffset) || 0);
+  } else {
+    leftOffset = TIMESERIES_CONSTANTS.gridOffsetLeft;
+  }
```

Comment:
This is maybe slightly heavy on the eyes. Would it be possible to add a few comments here explaining what's going on? Or better yet, maybe add a few self explanatory tests?